src/suggestion.rs

```diff
@@ -176,91 +167,75 @@ pub fn convert_long_statements_to_short(
-                        + PADDING_AROUND_LONG_LINES
-                        + padding_till_literal_start
-                        + 1,
-                )),
-        };
-        // Left range will not be used in this case
-        range_left_context = Range {
-            start: 0usize,
-            end: 0usize,
-        };
-        *offset = PADDING_OFFSET;
-    }
-    // left context has enough info to fill the terminal
-    // |---------left_context---------|-----misspelled_word-----|
-    //
-    // Attempt to fit the misspelled word with left context info
-    else if range_left_context.len() > terminal_size {
-        range_left_context = Range {
-            start: marker_range_relative
-                .start
-                .saturating_sub(terminal_size.saturating_sub(
-                    misspelled_word.chars().count()
-                        + PADDING_AROUND_LONG_LINES
-                        + padding_till_literal_start,
-                )),
-            end: marker_range_relative.start - 1,
-        };
-        // Right range will not be used
-        range_right_context = Range {
-            start: 0usize,
-            end: 0usize,
-        };
-        *offset = range_left_context.len() + PADDING_OFFSET;
-    }
-    // information will be shown in both sides of the `misspelled_word`
-    // |--left_context--|----misspelled_word---|--right_context--|
-    //
-    // Attempt to fit the misspelled word in the middle with info in th left and right of it
-    else {
-        let context = (terminal_size.saturating_sub(
-            misspelled_word.chars().count()
-                + padding_till_literal_start
-                + PADDING_AROUND_LONG_LINES,
-        )) / 2;
-        range_left_context = Range {
-            start: range_left_context.end.saturating_sub(context),
-            end: marker_range_relative.start - 1, //before the word starts
-        };
-        range_right_context = Range {
-            start: marker_range_relative.end - 1,
-            end: range_right_context.start + context,
-        };
-        *offset = range_left_context.len() + PADDING_OFFSET;
-    }
-    // Formatting itself added white spaces and punctuation to do the fitting to be considered:
-    //
-    //     |------ info ----| => PADDING_AROUND_LONG_LINES = 10 usize
-    // format!(
-    //     "  ... {}{}{} ...",
-    //     self.chunk.char_sub_window(range_left_context),
-    //     misspelled_word,
-    //     self.chunk.char_sub_window(range_right_context)
-    // )
+    let available_space = (terminal_size.saturating_sub(
+        misspelled_word.chars().count() + padding_till_literal_start + PADDING_AROUND_LONG_LINES,
+    )) / 2;
+    let mut left_context = Range {
+        start: 0,
+        end: range_word.start - 1,
+    };
+    let mut right_context = Range {
+        start: range_word.end,
+        end: stripped_line.chars().count(),
+    };
+    let left_remaining_space: i32 = available_space as i32 - left_context.len() as i32;
```

Comment:
there is also `isize`, if we need the sign.

Can you elaborate why we need signed types, from my understanding the context left and right can be of length zero in the "worst" case? Am I missing something?

Reply:
```
terminal_size
|-----------------------------------------------------------------------------|
|-------------||--------------------------------------------------------------|
(                                     total_available_space = 2*available_space
misspelled_word
+paddings
)

total_available_space
|--------------------------------------------------------------|

too long line
|------|---------------|----------------------------------------------------------------|

left_context      misspelled_word           right_context
```


ok, we have a lot of `right_context` but not so much `left_context`. But maybe some `left_context` contains 
import keywords such as `TODO` or `Important` etc. It would be cool if we try to show this as well, right?

Oki, let's try then...

```
    let left_remaining_space: i32 = available_space as i32 - left_context.len() as i32;
    let right_remaining_space: i32 = available_space as i32 - right_context.len() as i32;
```

In this case we would ended up with the case that 
`left_remaining_space>0` and `right_remaining_space<0` as `(true,false)` which means that we can use the whole 
`left_context `and reallocate remaining space to the right side. As well, the end of the right context is 
recalculated.

The same logic needs to follow the other cases as well.

Is this somehow helpful or is it more clear?

I think being explicit here is better to understand the logic and yeah, cast is annoying as here.

Reply:
That's all I wanted to understand, if the signs are a necessity or impl detail, but it's the latter. So I'd suggest to refactor this to get rid of all of the casts - makes things way easier to read and debug :)